webpack/assets/javascripts/react_app/components/AutoCompleteSearch/AutoCompleteActions.js

```diff
@@ -0,0 +1,50 @@
+import isEqual from 'lodash/isEqual';
+import API from '../../API';
+import { STATUS } from '../../constants';
+import { mapUsedQueryToPayload, getLastAction, urlBuilder } from './helpers';
+import {
+  AUTO_COMPLETE_REQUEST,
+  AUTO_COMPLETE_SUCCESS,
+  AUTO_COMPLETE_FAILURE,
+  AUTO_COMPLETE_RESET,
+  AUTO_COMPLETE_USED_QUERY,
+} from './AutoCompleteConstants';
+
+export const getResults = (path, searchQuery, controller, trigger) => (dispatch, getState) => {
+  const url = urlBuilder(path, searchQuery, trigger);
+  const items = {
+    url, searchQuery, controller, trigger,
+  };
+  const state = getState();
+  const payload = mapUsedQueryToPayload(controller, searchQuery, state);
+  if (payload) {
+    dispatch({
+      type: AUTO_COMPLETE_USED_QUERY,
+      payload,
+    });
+    return;
+  }
+  // Return early if this payload items are the same as in the last action.
+  const { items: lastItems } = getLastAction(state) || {};
+  if (lastItems && isEqual(lastItems, items)) {
+    return;
+  }
+  dispatch({ type: AUTO_COMPLETE_REQUEST, payload: { ...items, status: STATUS.PENDING } });
+
+  API.get(url)
+    .then(({ data }) => dispatch({
+      type: AUTO_COMPLETE_SUCCESS,
+      payload: { ...items, status: STATUS.RESOLVED, results: data },
+    }))
+    .catch(error => dispatch({
+      type: AUTO_COMPLETE_FAILURE,
+      payload: { error, ...items, status: STATUS.ERROR },
```

Comment:
Wait a second... why aren't you using `ajaxRequestAction`?
Why do you have to pass `STATUS.{ERROR,RESOLVED}` here? Most of the time we're doing this in the reducer.

Reply:
great question @ripcurld, it's because in `AUTO_COMPLETE_SUCCESS` `payload` we got the `results` mixed with all of the other items (  e.g: `{0,1,2,3, url, searchQuery, controller...}`), and needed to filter them and return only the results, not the items.

In this action I return the `data` that I've recieved from the `API call` as `results`.
Later in the reducer I just pick them as `payload.results` without the need to filter. :)